main/main.go

```diff
@@ -149,10 +144,10 @@ func main() {
 
 	// start the health check server
 	go func() {
-		log.Fatalln(http.ListenAndServe(fmt.Sprintf(":%d", defaultApiPort), nil))
+		log.Fatalln(http.ListenAndServe(fmt.Sprintf(":%d", cfg.APIPort), nil))
 	}()
 
-	startMetricsServer(logger, gatherer, defaultMetricsPort)
+	startMetricsServer(logger, gatherer, uint32(cfg.MetricsPort))
```

Comment:
can the type from config match one passed to `startMetricsServer`? Also, are ports valid up to `uint32`? From what I'm seeing online `uint16` would cover valid ports

Reply:
Yup, the max Unix port number is 2^16-1, so it could fit into a uint16.